### PR TITLE
Make DocService show fully qualified names when finding name conflicts

### PR DESCRIPTION
**Pull request by hyangtack**
Motivation:
Users may have multiple services, messages and/or enums with the same names but the different package names. However, DocService currently shows only their simple names in the UI, so it is not easy to distinguish them.

Modifications:
- Make DocService show the package name of the services, messages and/or enums when finding name conflicts.

Result:
- Users now can see the package name of their services, messages and enums with the same names from DocService.

### CODE REVIEW

**Comment by minwoox** [comment]
Where's the screenshot? 🤣

**Comment by hyangtack** [comment]
Added `example.grpc.hello.v2` package with the same `HelloService` to grpc-example.

image

File: docs-client/src/containers/App/index.tsx [hunk 1]
```diff
@@ -191,7 +192,14 @@ const AppDrawer: React.FunctionComponent<AppDrawerProps> = ({
...
-                    <Typography variant="subtitle1">
+                    <Typography display="inline" variant="body2">
+                      <code>
+                        {specification.hasUniqueServiceNames()
+                          ? ''
+                          : `${packageName(service.name)}.`}
+                      </code>
+                    </Typography>
+                    <Typography display="inline" variant="subtitle1">
                       <code>{simpleName(service.name)}</code>
```
Comment (minwoox) [inline docs-client/src/containers/App/index.tsx:203]: Would it be worth showing the full name when the mouse hovers when it has unique service names? 😆

**Review by minwoox — APPROVED** [review]
Awesome! Hope we can also implement #3977 so that users can customize the service name. 😆

**Review by ikhoon — APPROVED** [review]
Thanks, @hyangtack!

**Review by jrhee17 — APPROVED** [review]
Thank you! Looks great 😄 👍 🙇

File: docs-client/src/containers/App/index.tsx (hunk 1 above)
Comment (hyangtack) [reply docs-client/src/containers/App/index.tsx]: Added
image

Comment (hyangtack) [reply docs-client/src/containers/App/index.tsx]: @minwoox Would it be better if we show tooltip over struct/enum names as well?

Comment (minwoox) [reply docs-client/src/containers/App/index.tsx]: Yeah, I think it's useful. 😆

File: docs-client/src/containers/App/index.tsx [hunk 2]
```diff
@@ -191,8 +193,17 @@ const AppDrawer: React.FunctionComponent<AppDrawerProps> = ({
...
-                      <code>{simpleName(service.name)}</code>
+                    <Typography display="inline" variant="body2">
+                      <code>
+                        {specification.hasUniqueServiceNames()
+                          ? ''
+                          : `${packageName(service.name)}.`}
+                      </code>
+                    </Typography>
+                    <Typography display="inline" variant="subtitle1">
+                      <Tooltip title={`${service.name}`} placement="top">
```
Comment (ikhoon) [inline docs-client/src/containers/App/index.tsx:204]: ❤️

File: docs-client/src/containers/App/index.tsx (hunk 1 above)
Comment (hyangtack) [reply docs-client/src/containers/App/index.tsx]: @minwoox Please review https://github.com/line/armeria/pull/3979/commits/1ac876a7838a249e86646c630410ffe55d1a2ae8 again. I fixed a bug where the tooltip of the service name shows regardless whether there are unique names or not, and added tooltip for enum and struct. 🙇

Comment (minwoox) [reply docs-client/src/containers/App/index.tsx]: Thanks! Still LGTM. 👍

**Review by trustin — APPROVED** [review]
Nice work, @hyangtack :+1:

**Pull request merged**